### ISSUE TITLE
Provide opt-out for global implicit usings

### ISSUE THREAD

**Issue by jpobst** [SixLabors/ImageSharp#2430]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have verified that I am running the latest version of ImageSharp
- [X] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [X] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### ImageSharp version

3.0.1

### Other ImageSharp packages and versions

None

### Environment (Operating system, version and so on)

All

### .NET Framework version

.NET 7

### Description

There should be a way to opt out of using ImageSharp's global implicit usings, like `<DisableImageSharpImplicitUsings>` or something.

https://github.com/SixLabors/ImageSharp/blob/main/SixLabors.ImageSharp.props

```xml
<ItemGroup Condition="'$(ImplicitUsings)'=='enable' OR '$(ImplicitUsings)'=='true' AND '$(DisableImageSharpImplicitUsings)'!='true'">
```
Using global implicit usings in a winforms + ImageSharp project is very painful because there are many conflicts between `ImageSharp` and `System.Drawing`, like `Image`/`Size`/`SizeF`.

### Steps to Reproduce

None

**Comment by JimBobSquarePants** [comment]
See https://github.com/SixLabors/ImageSharp/discussions/2381#discussioncomment-5188759

**Comment by ds5678** [comment]
I also messaged on #2381.

If users want an implicit using, they can add it themselves. Putting your library in scope everywhere is ridiculous. No library (not even the BCL) should do that. It is an anti-pattern and should be reverted.

In the case of this issue, I think an opt-in is far more appropriate than an opt-out.

**Comment by ImrePyhvel** [comment]
### TL/DR:
I like the work you have been doing and happy to recommend ImageSharp so far, but I concur that:
this is a BAD decision. 

### Long version / rant ahead:

In our codebases a tiny 0.01% of it uses ImageSharp. But we do rely on `System.Drawing.Size` and `System.Drawing.Color` for other reasons. Upgrading ImageSharp to 3+  which was supposed to impact just 2 small services,  just magically broke much more important files which have **NOTHING to do with ImageSharp**.  And since the magic appearance of implicit usings is NOT intuitive nor expected (nor desired), some younger developers feel inclined to whack disambiguating using clauses everywhere. This is what happens in real world. 

* Yes, I know we could write more code to my project to suppress the feature we didn't want or need. But even that workaround is not intuitive, I don't even see a ImageSharp magic prop to disable "whatever Imagesharp did to my project setup". 
* We have carefully considered which namespaces we like being implicitly added. We have a internal pattern where and how we manage them, and ensure we keep them most helpful and without conflicts in OUR project. ImageSharp package does not know anything about our project or about other dependencies and should not impose its own questionable preferences on its consumers. 
* I believe as ImageSharp devs you may have a bias on how big part of all consumer code base builds solely around ImageSharp vs wider "some graphic tools", which rely on System.Drawing.*. I'm pretty sure System.Drawing.Size And System.Drawing.Color has a lot histsorical reach in real world consuming projects. That means there may be a lot more badly impacted devs and projects than you think. 

It is a good feature for internal packages. It MAY be a OK-ish feature, if all your type names are unique in 99+% of projects. Which is obviously **NOT** the case with Types like `Size` and `Color`. 

**This feature was not intuitive, not desired and it did waste our time for no gain. Please reconsider.**

**Comment by JimBobSquarePants** [comment]
I’m assuming here everyone has purchased a v3 license?

**Comment by kpko** [comment]
To add a viewpoint from a potential customer: I've just tried using ImageSharp in one of our projects because I'm evaluating libraries to validate images where I have a custom "Image" class (a database entity) and needed some time to find out why there are global usings added automatically in the build process. This was an awful experience and I never encountered another library that does that. As others said: This is *not* a positive change.

> I’m assuming here everyone has purchased a v3 license?

Would this make a difference? My experience was: Trying out the library, debugging the random using statements, looking for a workaround and found a closed discussion and this issue, where a project maintainer asked for a license. This doesn't give me much confidence in using a library, to be honest.

If we evaluated the library and come to the conclusion to buy a license, would you have considered removing the invasive global usings?

**Comment by JimBobSquarePants** [comment]
Fixed via https://github.com/SixLabors/ImageSharp/pull/2515
